### ISSUE TITLE
Raspberry Pi and Android Companion app error

### ISSUE THREAD

**Issue by apal21** [alexa/alexa-avs-sample-app#544]
I've installed this sample app on raspberry pi 3 B. It works perfectly fine while running javaclient but it throws and error on android companion app. I've tested my ip address by `ifconfig wlan0 | grep inet`. So what am I doing wrong?

![alexa companion app](https://cloud.githubusercontent.com/assets/11748050/23015620/8733c5da-f459-11e6-91ab-e6ddc9d1dc6f.jpg)

**Comment by elstryan** [comment]
Hi, 
This looks like you are running on an actual Android device. Can you try running on an emulator first following the steps in our guide? https://github.com/alexa/alexa-avs-sample-app/wiki/Mac#5b---build-and-run-an-android-companion-app-authentication 
This way we can narrow down where the problem might be occurring.

**Comment by apal21** [comment]
Okay I'll try this in Android studio today nd tell you if there is an error

**Comment by apal21** [comment]
It works on Android studio (Windows) but not working if the code is running on the raspberry pi

**Comment by yugoren** [comment]
Hi @apal21,

I have a couple of points where I think the problem might have occurred:
1. Have you tried enabling only WiFi on your Android Device? If the authorization is trying to go over LTE, it wouldn't connect to the IP.
2. Have you tried pinging 8443 on RPi and/or checked the configuration that it's accepting outside connection to that port? I'm not an Android developer but the emulator might expose that port automatically while you might need more setup for the actual port.

Please let us know about your progress with any questions you might have.

**Comment by apal21** [comment]
Yes I was on the same network so IP address is not an issue and yes I can ping port 8443 but I don't understand why it shows this issue.
I cannot connect through any external IP. So may be the ext IP is blocked somewhere in the raspberry pi's alexa

**Comment by jordan-heier** [comment]
Hi @apal21,

Can you confirm that you have added the IP address that you are using as an alternate name in the ssl.cnf https://github.com/alexa/alexa-avs-sample-app/blob/master/samples/javaclient/ssl.cnf

If you haven't done this, you will need to add the IP address you are using and regenerate the certificates.

**Comment by apal21** [comment]
No I haven't added that. I'll try after adding it.

**Comment by shaolinkhoa** [comment]
Hi @jordan-heier ,

I don't understand in step 4 - linux. If I use an Android app from a devide, Which entry I have to add to `[alt_names]` in `ssl.cnf` ? A IP address of my phone while connecting wifi or what ?

**Comment by dskidmor** [comment]
The `ssl.cnf` IP address will refer to the raspberry pi. If you need help finding the address, try running `ifconfig` from the command line.
